### PR TITLE
Implement private tally certificate in jcli

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Implements #2770

### CODE REVIEW

File: jormungandr-lib/src/interfaces/vote.rs [hunk 1]
```diff
@@ -471,6 +471,23 @@ mod serde_base64_bytes {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TallyDecryptShare(#[serde(with = "serde_base64_bytes")] Vec<u8>);
```
Comment (mzabaluev) [inline jormungandr-lib/src/interfaces/vote.rs:475]: This structure is not used in the API. Maybe make it internal for jcli for the time being?..

File: jormungandr-lib/src/interfaces/vote.rs [hunk 2]
```diff
@@ -471,6 +471,23 @@ mod serde_base64_bytes {
...
+
+impl TryFrom<TallyDecryptShare> for chain_vote::TallyDecryptShare {
+    // TODO: Maybe change this error to a custom type?
+    type Error = std::io::Error;
```
Comment (mzabaluev) [inline jormungandr-lib/src/interfaces/vote.rs:479]: ... then you can use `certificate::Error` for the command implementation.

This impl looks atypical for what the rest of jormungandr-lib does, let's not add it here until it's clear that we also need the serde impls for REST.

File: jcli/src/jcli_app/certificate/mod.rs [hunk 3]
```diff
@@ -98,6 +98,8 @@ pub enum Error {
     VoteEncryptingKey,
     #[error("invalid bech32 public key, expected {expected} hrp got {actual}")]
     InvalidBech32Key { expected: String, actual: String },
+    #[error("Invalid json representation")]
```
Comment (eugene-babichenko) [inline jcli/src/jcli_app/certificate/mod.rs:101]: Maybe something like "Invalid shares JSON" to make it clear?

File: jormungandr-lib/src/interfaces/vote.rs [hunk 4]
```diff
@@ -405,7 +405,7 @@ impl TallyResult {
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct EncryptedTally(#[serde(with = "serde_base64_bytes")] Vec<u8>);
 
-mod serde_base64_bytes {
+pub mod serde_base64_bytes {
```
Comment (danielSanchezQ) [inline jormungandr-lib/src/interfaces/vote.rs:408]: I think we should start to move this kind of utilities to some other shared module ('serde_utils' or whatever). We can discuss it and I'll do it in a further PR.

File: testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs [hunk 5]
```diff
@@ -27,9 +27,9 @@ impl Certificate {
             .as_single_line()
     }
 
-    pub fn new_vote_tally<S: Into<String>>(self, vote_plan_id: S) -> String {
+    pub fn new_vote_tally<S: Into<String>>(self, vote_plan_id: S, tally_type: S) -> String {
```
Comment (danielSanchezQ) [inline testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs:30]: @dkijania are these changes ok? or do you think it should have a custom type somewhere?

File: jcli/src/jcli_app/certificate/mod.rs [hunk 6]
```diff
@@ -98,6 +98,10 @@ pub enum Error {
     VoteEncryptingKey,
     #[error("invalid bech32 public key, expected {expected} hrp got {actual}")]
     InvalidBech32Key { expected: String, actual: String },
+    #[error("Invalid shares JSON representation")]
+    InvalidJson(#[from] serde_json::Error),
+    #[error("Invalid binary share")]
```
Comment (mzabaluev) [inline jcli/src/jcli_app/certificate/mod.rs:103]: This should be clearer worded, I think: "invalid binary share data"?
Capitalization should be more consistent across the enum. Let's go with small letters everywhere.

File: jormungandr-lib/src/interfaces/vote.rs (hunk 4 above)
Comment (mzabaluev) [reply jormungandr-lib/src/interfaces/vote.rs:408]: Yeah, but check also https://crates.io/crates/base64-serde

File: testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs (hunk 5 above)
Comment (mzabaluev) [reply testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs:30]: A custom type would be more disciplined. Could reuse from jormungandr-lib?

File: jcli/src/jcli_app/certificate/mod.rs (hunk 6 above)
Comment (danielSanchezQ) [reply jcli/src/jcli_app/certificate/mod.rs:103]: my bad should have realized it :/

File: jormungandr-lib/src/interfaces/vote.rs (hunk 4 above)
Comment (danielSanchezQ) [reply jormungandr-lib/src/interfaces/vote.rs:408]: I knew about that library. But, without having checked it properly maybe. I would say it doesn't support the check we do for the human-readable serializer in there.

File: testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs (hunk 5 above)
Comment (danielSanchezQ) [reply testing/jormungandr-integration-tests/src/common/jcli/api/certificate.rs:30]: We could use `PayloadTypeDef`..

File: testing/jormungandr-integration-tests/src/common/jcli/command/certificate.rs [hunk 7]
```diff
@@ -163,3 +168,11 @@ impl CertificateCommand {
         self.command
     }
 }
+
+fn payload_type_to_string(payload_type: PayloadType) -> String {
```
Comment (mzabaluev) [inline testing/jormungandr-integration-tests/src/common/jcli/command/certificate.rs:172]: No need to allocate, just return `&'static str`.

**Review by mzabaluev — COMMENTED** [review]
The rest looks OK, just fix the tests that are failing to build.

**Review by mzabaluev — APPROVED** [review]

**Pull request merged**